pkg/isbsvc/jetstream_service.go

```diff
@@ -114,7 +114,7 @@ func (jss *jetStreamSvc) CreateBuffersAndBuckets(ctx context.Context, buffers, b
 				Name:       streamName,
 				Subjects:   []string{streamName}, // Use the stream name as the only subject
 				Retention:  nats.RetentionPolicy(v.GetInt("stream.retention")),
-				Discard:    nats.DiscardOld,
+				Discard:    nats.DiscardNew,
```

Comment:
can we make it overridable by the user?

Reply:
Maybe we can set it when users specify "drop on full" ? It's pretty much the exact same behaviour.
Though according to @yhl25 DiscardNew can't be used, but I'm still trying to understand why in the other issue.

Reply:
+1 to make it configurable to the user, but let's set it to `DiscardOld` by default since we use the `Limits` Policy by default. Also, please add a comment saying `DiscardNew` can only be used with `WorkQueue` Policy.

Reply:
since we cannot use `DiscardNew` with `Limits` policy. we should not let the pipeline even start and validation should fail.

Reply:
Maybe we should also document the risk of data loss on surge when using DiscardOld on high throughput and be really transparent with why it happens. Right now, users like me who fiddle with the config might be in big trouble if some UDF/Sink create a silent data loss scenario on production.